omnigibson/prims/xform_prim.py

```diff
@@ -168,53 +169,114 @@ def has_material(self):
+                rotq = lazy.pxr.Gf.Quatd(*orientation)
+            xform_op.Set(rotq)
+            PoseAPI.invalidate()
+            if gm.ENABLE_FLATCACHE:
+                # If flatcache is on, make sure the USD local pose is synced to the fabric local pose.
+                # Ideally we should call usdrt's set local pose directly, but there is no such API.
+                # The only available API is SetLocalXformFromUsd, so we update USD first, and then sync to fabric.
+                xformable_prim = lazy.usdrt.Rt.Xformable(
+                    lazy.omni.isaac.core.utils.prims.get_prim_at_path(self.prim_path, fabric=True)
+                )
+                assert (
+                    not xformable_prim.HasWorldXform()
+                ), "Fabric's world pose is set for a non-rigid prim which is unexpected. Please report this."
+                xformable_prim.SetLocalXformFromUsd()
 
-    def get_position_orientation(self, clone=True):
+    def get_position_orientation(self, frame: Literal["world", "scene", "parent"] = "world", clone=True):
         """
-        Gets prim's pose with respect to the world's frame.
+        Gets prim's pose with respect to the specified frame.
+
+        Args:
+            frame (Literal): frame to get the pose with respect to. Default to world. parent frame
+            get position relative to the object parent. scene frame get position relative to the scene.
+            clone (bool): Whether to clone the internal buffer or not when grabbing data
 
         Args:
             clone (bool): Whether to clone the internal buffer or not when grabbing data
 
         Returns:
             2-tuple:
-                - 3-array: (x,y,z) position in the world frame
-                - 4-array: (x,y,z,w) quaternion orientation in the world frame
+                - th.Tensor: (x,y,z) position in the specified frame
+                - th.Tensor: (x,y,z,w) quaternion orientation in the specified frame
         """
-        return PoseAPI.get_world_pose(self.prim_path)
+
+        assert frame in ["world", "parent", "scene"], f"Invalid frame '{frame}'. Must be 'world', 'parent', or 'scene'."
+
+        if frame == "world" or frame == "scene":
+            position, orientation = PoseAPI.get_position_orientation(self.prim_path, "world")
+
+            # If we are in a scene, compute the scene-local transform (and save this as the world transform
+            # for legacy compatibility)
+            if frame == "scene":
+                if self.scene is None:
+                    raise ValueError("Cannot get position and orientation relative to scene without a scene")
+                else:
+                    position, orientation = T.mat2pose(self.scene.pose_inv @ T.pose2mat((position, orientation)))
+
+            return position, orientation
+        else:
+            return PoseAPI.get_position_orientation(self.prim_path, frame="parent")
+
+    # ------------------- Deprecated methods -------------------
```

Comment:
Remove this comment. Also remove emptylines from all of the block beginnings below.